src/bin/sol-fbp-runner/Port.js

```diff
@@ -0,0 +1,122 @@
+(function(window) {
+
+	'use strict'
+
+	function Port (params) {
+
+		//data properties
+		var instance = this; //returns the __proto__ of the object
+
+		this.name = params["name"];
+		this.value = params["value"];
+		this.type = params["type"] || "in";
+		this.data_type = params["data_type"];
+		this.widget = params["widget"];
+		this.port_index = params["port_index"];
+		this.required = params["required"] || true;
+		this.key = "port_" + this.type + "_" + params["key"];
```

Comment:
All these parameters should be written in dot notation instead of using ['key']

It would be something like this:
```
this.required = params.required;
```

You can see more info about this here:
https://jslinterrors.com/a-is-better-written-in-dot-notation